### DISCUSSION TITLE
I have an issue with Laravel config:cache and SQS queues

### DISCUSSION THREAD

**Discussion by vedmant** [brefphp/bref#1881] [General]
I have an issue with Laravel `config:cache` and SQS queues. I used approach with storing secrets in the github and generating config cache in github actions, this approach looked ok since config:cache was recommended in docs for faster cold starts. However since `SQS_QUEUE: ${construct:jobs.queueUrl}` is passed from the lambda environment, Laravel will not read this since config is already cached. 

The question is how I can then pass `SQS_QUEUE: ${construct:jobs.queueUrl}` to Laravel configuration keeping pre cached config with all secrets?

**Comment by mnapoli** [comment]
Hi, that's a good question to be honest, I'm not sure on the top of my head whether:

1. config caching will hardcode the current `SQS_QUEUE` value everywhere
2. or it will only hardcode it **if it is set**

So I would try this: do not define `SQS_QUEUE` in the `.env` that is used for deployment, so that it is not cached (in case it's `2.`).

If it's `1.`, then I'm afraid pre-caching doesn't work in these cases? i.e. the `SQS_QUEUE` value needs to be set in `.env`.

That's interesting that no one brought this up yet.

**Reply by vedmant** [reply]
@mnapoli Thanks, Laravel doesn't load any envs into config when it's cached. I just came up with a simple workaround:
```php
// app/Providers/AppServiceProvider.php
    public function boot()
    {
        config()->set('queue.connections.sqs.queue', env('SQS_QUEUE'));
    }
```
